src/Compress/Bz2Adapter.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\Compress;
+
+use Laminas\Filter\Exception\ExtensionNotLoadedException;
+use Laminas\Filter\Exception\RuntimeException;
+
+use function assert;
+use function bzcompress;
+use function bzdecompress;
+use function extension_loaded;
+use function is_int;
+use function sprintf;
+
+/**
+ * Compression adapter for Bz2
+ *
+ * @psalm-type Options = array{
+ *     blocksize?: int<1, 9>|null,
+ * }
+ */
+final class Bz2Adapter implements StringCompressionAdapterInterface
```

Comment:
The BZ2 adapter now only supports string content.

In future, it will be possible to introduce a `FileCompressionInterface` that this adapter _could_ also implement so that we can filter an arbitrary string, or a single un-compressed file to `somefile.bz2`